zha/application/platforms/update.py

```diff
@@ -231,16 +238,41 @@ def handle_cluster_handler_attribute_updated(
             self._attr_installed_version = f"0x{event.attribute_value:08x}"
             self.maybe_emit_state_changed_event()
 
-    def device_ota_update_available(
-        self, image: OtaImageWithMetadata, current_file_version: int
+    def device_ota_image_query_result(
+        self,
+        images_result: OtaImagesResult,
+        query_next_img_command: QueryNextImageCommand,
     ) -> None:
         """Handle ota update available signal from Zigpy."""
-        self._latest_firmware = image
-        self._attr_latest_version = f"0x{image.version:08x}"
-        self._attr_installed_version = f"0x{current_file_version:08x}"
 
-        if image.metadata.changelog:
-            self._attr_release_summary = image.metadata.changelog
+        current_version = query_next_img_command.current_file_version
+        self._attr_installed_version = f"0x{current_version:08x}"
+
+        self._compatible_images = images_result
+        self._attr_latest_version = None
+        self._attr_release_summary = None
+        self._attr_release_notes = None
+
+        latest_firmware: OtaImageWithMetadata | None = None
+
+        if images_result.upgrades:
+            # If there are upgrades, cache the image and indicate that we should upgrade
+            latest_firmware = images_result.upgrades[0]
+            self._attr_latest_version = f"0x{latest_firmware.version:08x}"
+            self._attr_release_summary = (
+                latest_firmware.metadata.changelog
+                if latest_firmware.metadata.changelog
+                else None
+            )
+            self._attr_release_notes = (
+                latest_firmware.metadata.release_notes
+                if latest_firmware.metadata.release_notes
+                else None
+            )
```

Comment:
We might be able to do this:
```suggestion
            self._attr_release_summary = latest_firmware.metadata.changelog or None
            self._attr_release_notes = latest_firmware.metadata.release_notes or None
```